include/triton/Dialect/TritonGPU/IR/TritonGPUInterfaces.h

```diff
@@ -2,11 +2,23 @@
 #define TRITON_GPU_DIALECT_INTERFACES_H
 
 #include "mlir/IR/OpDefinition.h"
+#include "mlir/Interfaces/SideEffectInterfaces.h"
 
 // clang-format off
 #include "triton/Dialect/TritonGPU/IR/LinearLayoutConversions.h"
 #include "triton/Dialect/TritonGPU/IR/OpInterfaces.h.inc"
 #include "triton/Dialect/TritonGPU/IR/AttrInterfaces.h.inc"
 // clang-format on
 
+namespace mlir::MemoryEffects {
```

Comment:
Can you add more comments to describe why they are "atomic"?

Also, is `atomic` the correct terminology to describe the effect?

I understand that these operations are internally "synced" as only a single thread will perform them, but this isn't the same as "atomic".

Reply:
> It simply means, when multiple threads/warps do mbarrier.arrive/expect_tx/complet_tx without other ordering/"at the same time", there will be no hazard.

It's due to the fact that these instructions are executed per warp group basis. In that sense it's kind of "atomic" to each warp group but I'm still not sure it describes the behavior correctly. **Can you add some more comments in the code in either here or `Membar.h`.**

> BTW per my experiment, mbarrier.arrive/expect_tx are not issued on a per-warp basis, only tmp load/store are issued on a per-warp basis. 

Yes